### PR TITLE
Add more native docker test cases

### PR DESCRIPTION
**Pull request by cjellick**
Add test cases for container exec, logs, and ip injection

### CODE REVIEW

**Comment by sonchang** [comment]
LGTM

File: tests/validation/cattlevalidationtest/core/test_container.py [hunk 1]
```diff
@@ -117,30 +117,26 @@ def test_linking(client, test_name, managed_network):
...
-
-def cleanup(container):
-        try:
-            container.stop(remove=True, timeout=0)
-        except:
-            # We've tried our best!
-            pass
+def test_ip_inject(client, managed_network, test_name):
+    cleanup_items = []
+    try:
```
Comment (alena1108) [inline tests/validation/cattlevalidationtest/core/test_container.py]: @cjellick Craig, question about ip allocation. Who is responsible for handing out the free ip address - python-agent or cattle server? And if its implemented on both sides with python-agent doing it for native containers, and cattle server for rancher ones, how do we do cross check for ip conflicts?

Comment (cjellick) [inline tests/validation/cattlevalidationtest/core/test_container.py]: It is ***always*** cattle server. With native containers, the user must add an environment variable to the container `RANCHER_NETWORK=true` (will be a label in the future). When the native container's existence is reported to cattle, cattle will allocate an IP as part of the normal container create process if that env var is present.

Comment (alena1108) [inline tests/validation/cattlevalidationtest/core/test_container.py]: Got it, thank you!

**Comment by alena1108** [comment]
LGTM

**Pull request merged**